types/lib/model.d.ts

```diff
@@ -287,44 +291,53 @@ export interface WhereOperators {
    *
    * Example: `[Op.notIRegexp]: '^[h|a|t]'` becomes `!~* '^[h|a|t]'`
    */
-  [Op.notIRegexp]?: string;
+  [Op.notIRegexp]?: WhereOperators[typeof Op.regexp];
 
   /**
    * PG only
    *
    * Forces the operator to be strictly left eg. `<< [a, b)`
    */
-  [Op.strictLeft]?: Rangable;
+  [Op.strictLeft]?: AllowOrAnd<Rangable>;
 
   /**
    * PG only
    *
    * Forces the operator to be strictly right eg. `>> [a, b)`
    */
-  [Op.strictRight]?: Rangable;
+  [Op.strictRight]?: WhereOperators[typeof Op.strictLeft];
 
   /**
    * PG only
    *
    * Forces the operator to not extend the left eg. `&> [1, 2)`
    */
-  [Op.noExtendLeft]?: Rangable;
+  [Op.noExtendLeft]?: WhereOperators[typeof Op.strictLeft];
 
   /**
    * PG only
    *
    * Forces the operator to not extend the left eg. `&< [1, 2)`
    */
-  [Op.noExtendRight]?: Rangable;
-
+  [Op.noExtendRight]?: WhereOperators[typeof Op.strictLeft];
 }
 
-/** Example: `[Op.or]: [{a: 5}, {a: 6}]` becomes `(a = 5 OR a = 6)` */
+/**
+ * Example: `[Op.or]: [{a: 5}, {a: 6}]` becomes `(a = 5 OR a = 6)`
+ *
+ * @deprecated do not use me!
+ */
+// TODO [>6]: Remove me
 export interface OrOperator<TAttributes = any> {
   [Op.or]: WhereOptions<TAttributes> | readonly WhereOptions<TAttributes>[] | WhereValue<TAttributes> | readonly WhereValue<TAttributes>[];
 }
 
-/** Example: `[Op.and]: {a: 5}` becomes `AND (a = 5)` */
+/**
+ * Example: `[Op.and]: {a: 5}` becomes `AND (a = 5)`
+ *
+ * @deprecated do not use me!
+ */
+// TODO [>6]: Remove me
```

Comment:
Does this note apply to v7 or v8? 🤔

Reply:
It should be removed in v7 but since I don't want to cherry pick the removal in v6 I'll do it in a follow-up PR